test/widgets/home_test.dart

```diff
@@ -0,0 +1,409 @@
+      checkOnLoadingPage();
+    }
+
+    Future<void> tapChooseAccount(WidgetTester tester) async {
+      await tester.tap(find.text('Try another account'));
+      await tester.pump(Duration.zero); // tap the button
+      await tester.pump(const Duration(milliseconds: 250)); // wait for animation
+      checkOnChooseAccountPage();
+    }
+
+    Future<void> chooseAccountWithEmail(WidgetTester tester, String email) async {
+      await tester.tap(find.text(email));
+      await tester.pump(Duration.zero); // tap the button
+      await tester.pump(const Duration(milliseconds: 350)); // wait for push & pop animations
+      checkOnLoadingPage();
+    }
+
+    testWidgets('smoke', (tester) async {
+      addTearDown(testBinding.reset);
+      testBinding.globalStore.loadPerAccountDuration = loadPerAccountDuration;
+      await prepare(tester);
+      await tester.pump(loadPerAccountDuration);
+      checkOnHomePage(tester, expectedAccount: eg.selfAccount);
+    });
+
+    testWidgets('"Try another account" button appears after timeout', (tester) async {
+      addTearDown(testBinding.reset);
+      testBinding.globalStore.loadPerAccountDuration = loadPerAccountDuration;
+      await prepare(tester);
+      checkOnLoadingPage();
+      check(find.text('Try another account').hitTestable()).findsNothing();
+
+      await tester.pump(kTryAnotherAccountWaitPeriod);
+      checkOnLoadingPage();
+      check(find.text('Try another account').hitTestable()).findsOne();
+
+      await tester.pump(loadPerAccountDuration);
+      checkOnHomePage(tester, expectedAccount: eg.selfAccount);
+    });
+
+    testWidgets('while loading, go back from ChooseAccountPage', (tester) async {
+      testBinding.globalStore.loadPerAccountDuration = loadPerAccountDuration;
+      await prepare(tester);
+      await tester.pump(kTryAnotherAccountWaitPeriod);
+      await tapChooseAccount(tester);
+
+      await tester.tap(find.byType(BackButton));
+      await tester.pump(Duration.zero); // tap the button
+      await tester.pump(const Duration(milliseconds: 350)); // wait for pop animation
+      checkOnLoadingPage();
+
+      await tester.pump(loadPerAccountDuration);
+      checkOnHomePage(tester, expectedAccount: eg.selfAccount);
+    });
+
+    testWidgets('while loading, choose a different account', (tester) async {
+      testBinding.globalStore.loadPerAccountDuration = loadPerAccountDuration;
+      await prepare(tester);
+      await tester.pump(kTryAnotherAccountWaitPeriod);
+      await tapChooseAccount(tester);
+
+      testBinding.globalStore.loadPerAccountDuration = loadPerAccountDuration * 2;
+      await chooseAccountWithEmail(tester, eg.otherAccount.email);
+
+      // The second loadPerAccount is still pending.
+      await tester.pump(loadPerAccountDuration);
+      checkOnLoadingPage();
+
+      // The second loadPerAccount finished.
+      await tester.pump(loadPerAccountDuration);
+      checkOnHomePage(tester, expectedAccount: eg.otherAccount);
```

Comment:
nit: comment that says something is true, in the present tense, should go at a point where it's become true

```suggestion
      await tester.pump(loadPerAccountDuration);
      // The second loadPerAccount finished.
      checkOnHomePage(tester, expectedAccount: eg.otherAccount);
```